docs/docs/howto/encrypted-storage.md

```diff
@@ -9,177 +9,152 @@ This step is recommended for any Contrast deployment that stores sensitive data
+Give the device a name (the example uses `device-name`).
+
+Finally, add the [Contrast annotation](../architecture/k8s-yaml-elements.md) `contrast.edgeless.systems/secure-pv` with value `device-name:mount-name`.
+
+```yaml
+spec: # v1.StatefulSetSpec
+  template:
+    metadata:
+      annotations:
+        contrast.edgeless.systems/secure-pv: device-name:mount-name
+    spec:
+      containers:
+      - volumeMounts:
+        - mountPath: /mount/path
+          mountPropagation: HostToContainer
+          name: mount-name
+      volumes:
+      - emptyDir: {}
+        name: mount-name
+  volumeClaimTemplates:
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: device-name
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 1Gi
+      volumeMode: Block
+```
+
+The presence of this annotation instructs an init container running `cryptsetup`
+to use the workload secret at `/contrast/secrets/workload-secret-seed` to generate
+a key and setup the block device as a LUKS partition. Before starting the application container,
+the init container uses the generated key to open the LUKS device, which is then mounted
+by the application container. For the application container, this process is completely
+transparent and works like mounting any other volume. The `cryptsetup` container
+will remain running to provide the necessary decryption context for the workload
+container.
 
-:::note[Attention!]
+:::note[Persistent workload secrets]
 
-This must be done on a trusted machine, with a secure and trusted connection to it.
+During the initialization process of the workload pod, the Contrast Initializer
+sends an attestation report to the Coordinator and receives a [workload secret](../architecture/secrets.md#workload-secrets)
+derived from the Coordinator's secret seed and the workload secret ID specified in the
+manifest, and writes it to a secure in-memory `volumeMount`.
 
 :::
 
-</TabItem>
-<TabItem value="metal-qemu-snp-gpu" label="Bare metal (SEV-SNP, with GPU support)">
-
-```sh
-contrast generate --reference-values metal-qemu-snp-gpu resources/
-```
+### Applying the configuration
 
-On bare-metal SEV-SNP, `contrast generate` is unable to fill in the `MinimumTCB` values as they can vary between platforms and CPU models.
-They will have to be filled in manually.
-
-If you don't know the values from the firmware you installed, you can use the [`snphost`](https://github.com/virtee/snphost) tool to retrieve the current TCB.
+To make the changes take effect, reapply your deployment:
 
 ```sh
-snphost show tcb
-```
-```console
-Reported TCB: TCB Version:
-  Microcode:   72
-  SNP:         23
-  TEE:         0
-  Boot Loader: 9
-  FMC:         None
-Platform TCB: TCB Version:
-  Microcode:   72
-  SNP:         23
-  TEE:         0
-  Boot Loader: 9
-  FMC:         None
+contrast generate resources/
+kubectl apply -f resources/
+contrast set resources/
```

Comment:
This is not executable (missing coordinator address). Instead of writing the commands down here, should we just point to the other how-tos?